src/gui/MainWindow.cpp

```diff
@@ -1034,8 +1034,8 @@ void MainWindow::updateWindowTitle()
     if (stackedWidgetIndex == DatabaseTabScreen && tabWidgetIndex != -1) {
         customWindowTitlePart = m_ui->tabWidget->tabName(tabWidgetIndex);
         if (isModified) {
-            // remove asterisk '*' from title
-            customWindowTitlePart.remove(customWindowTitlePart.size() - 1, 1);
+            // Remove asterisk '*' from title
+            customWindowTitlePart.replace("*", "");
```

Comment:
This will technically remove all asterisks in the tab name entirely which may be totally valid if the database name has them like `**MY DATABASE**`. The previous code was actually "right" but relied on DatabaseTabWidget doing what we expect.

Reply:
Didn't though about that kind of naming scenario. I'll modify this to a regex, or whatever suits the best.